### ISSUE TITLE
Compilation fails on `prepareDokkatooParametersJavadoc` on gradle 8.1.1

### ISSUE THREAD

**Issue by Nek-12** [adamko-dev/dokkatoo#82]
```
Some problems were found with the configuration of task ':project:prepareDokkatooParametersJavadoc' (type 'DokkatooPrepareParametersTask').
  - Gradle detected a problem with the following location: '/project/.gradle/kotlin/kotlinTransformedCInteropMetadataLibraries/org.jetbrains.kotlinx-atomicfu-0.20.2-nativeMain-cinterop/org.jetbrains.kotlinx_atomicfu-cinterop-interop-nfvfsw.klib'.
    
    Reason: Task ':apiresult:prepareDokkatooParametersJavadoc' uses this output of task ':project2:transformIosMainCInteropDependenciesMetadataForIde' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':project2:transformIosMainCInteropDependenciesMetadataForIde' as an input of ':project:prepareDokkatooParametersJavadoc'.
      2. Declare an explicit dependency on ':project2:transformIosMainCInteropDependenciesMetadataForIde' from ':project:prepareDokkatooParametersJavadoc' using Task#dependsOn.
      3. Declare an explicit dependency on ':project2:transformIosMainCInteropDependenciesMetadataForIde' from ':project:prepareDokkatooParametersJavadoc' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.

```
gradle.properties includes:
```properties
kotlin.mpp.enableCInteropCommonization=true
kotlin.mpp.stability.nowarn=true
org.gradle.unsafe.configuration-cache=true
```

**Comment by Nek-12** [comment]
Disabling C InteropCommonization fixes the issue

**Comment by aSemy** [comment]
Hey, thanks for the report.

By my understanding this is a Kotlin Gradle Plugin issue. KGP is defining a Gradle Task `:project2:transformIosMainCInteropDependenciesMetadataForIde` but not properly declaring the output files, which leads to Gradle having issues.

I'm reluctant to try putting a workaround in the codebase, but perhaps the workaround you've found could be documented.

As well as disabling commonization, could you try the workaround in the Redwood project and report back?

https://github.com/cashapp/redwood/blob/3be2d1b3947866c5e16008823ff1eacaef627823/build.gradle#L100-L113

**Comment by Nek-12** [comment]
Yes, I have tried it, it does not produce consistent results for my purposes (javadocs are produced during compilation as a dependency of `publishToMavenLocal`). But thanks for your response.

**Comment by Nek-12** [comment]
I'm not sure if this issue should be open here, but at a second thought, I'll keep it open just in case. Feel free to close this as you see fit

**Comment by aSemy** [comment]
For now I'm happy to leave this open in lieu of writing proper documentation. I think that this problem with be fixed with Kotlin 1.9.0, as the Kotlin Gradle Plugin should be completely compatible with Gradle Config cache.
